internal/editions.go

```diff
@@ -16,10 +16,20 @@ package internal
 
 import "google.golang.org/protobuf/types/descriptorpb"
 
-// AllowEditions is set to true in tests to enable editions syntax for testing.
-// This will be removed and editions will be allowed by non-test code once the
+var allowEditions = false
+
+// AllowEditionsForTest is called to enable editions syntax for testing. This
+// will be removed and editions will be allowed by non-test code once the
 // implementation is complete.
-var AllowEditions = false
+func AllowEditionsForTest() {
```

Comment:
I think this can be done directly for the bool? See as an example https://github.com/google/starlark-go/blob/e6e8e7ce1b7a8dff097a7e16bdacc53590e942ae/syntax/options.go#L56
Although is it read only?

Reply:
Oh, good point. All of the articles/posts/etc I've found only talk about swapping out functions. But I now see what appears to be the "canonical docs" in https://pkg.go.dev/cmd/compile, which mention using it for vars as well.

In that case, I can go ahead and close this PR as no changes are needed for such "sneaky" use of editions from code outside of this repo's tests.